source/features/wait-for-attachments.ts

```diff
@@ -0,0 +1,43 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+function toggleSubmitButtons(target: HTMLElement, disabled: boolean): void {
+	const textarea = select('textarea', target)!;
+	if (disabled) {
+		textarea.removeAttribute('data-required-trimmed');
+	} else {
+		textarea.dataset.requiredTrimmed = 'Text field is empty';
+	}
+
+	// Force the "Update comment"/"Create pull request" buttons to be disabled
+	for (const button of select.all('button[type="submit"]', target.closest('form')!)) {
+		button.toggleAttribute('disabled', disabled);
+	}
```

Comment:
One of these is unnecessary. Either you toggle the `data-required-trimmed` attribute _or_ manually disable the buttons. The whole point of removing `data-required-trimmed` would be so that we don't have to manually disable/enable the buttons since that's handled by the upload events

Reply:
That's what I thought too, but for some reason the "Comment" button doesn't get disabled if the `data-required-trimmed` attribute is not removed on the textarea. My guess is that the `disabled` attribute on the button is silently overwritten by GitHub at some point before or during the upload.

So the attribute removal is for the "Comment" button on new comment forms, and the `for` loop for all the others (including the ones added by RGH).

Reply:
> the `for` loop for all the others

I'd rather address the source of this problem directly, which incidentally is the same reason why "Close with comment" is also unaffected:

`data-disable-invalid` isn't set on either button.

I made some changes but I did not address this issue specifically yet.

Reply:
I think `data-disable-invalid` can't be set on "Close with comment" because the same button has to work as a plain "Close pull request" when the field is empty.

For this button we do indeed need the `for` loop, so we might as well restore it for both comment editing and "close with comment" buttons.

Reply:
> need the `for` loop, so we might as well restore it for both comment editing and "close with comment" buttons.

This is still missing

Reply:
> This is still missing

Done. I also had to change the selector for the textarea because the previous one didn't work when editing a comment or on the new PR page.